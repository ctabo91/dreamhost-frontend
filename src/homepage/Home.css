.Home-title {
    font-family: "Aboreto", system-ui;
    font-weight: 400;
    font-style: normal;
}

.logo {
    font-family: "Satisfy", cursive;
    font-weight: 400;
    font-style: normal;
}