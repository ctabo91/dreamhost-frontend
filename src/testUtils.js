import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser, hasFavoritedRecipe = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasFavoritedRecipe }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };