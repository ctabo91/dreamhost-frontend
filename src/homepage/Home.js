import React, { useContext } from "react";
import UserContext from "../auth/UserContext";
import "./Home.css";

/** Home page for App.
 * 
 * Shows a welcome message, depending on whether or not user is logged in.
 */ 

function Home() {
    const { currentUser } = useContext(UserContext);

    return (
        <div className="Home mt-5 pt-5">
            <h1 className="Home-title display-1 text-center pt-5">
                {
                    !currentUser
                        ? <>
                            <span>Welcome To</span>
                            <p className="logo">DreamHost</p>
                          </>
                        : `Welcome ${currentUser.firstName}!`
                }
            </h1>
        </div>
    );
}

export default Home;