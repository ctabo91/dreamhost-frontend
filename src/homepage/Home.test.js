import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "./Home";
import { UserProvider } from "../testUtils";

describe("Home Component", () => {
    it("renders without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <Home />
                </UserProvider>
            </MemoryRouter>,
        );
    });
    
    it("matches snapshot", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider>
                    <Home />
                </UserProvider>
            </MemoryRouter>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    
    it("matches snapshot when logged out", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider currentUser={null}>
                    <Home />
                </UserProvider>
            </MemoryRouter>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    
    it("shows correct welcome message when logged in", () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <Home />
                </UserProvider>
            </MemoryRouter>,
        );
    
        expect(screen.getByText('Welcome testfirst!')).toBeInTheDocument();
    });
    
    it("shows correct welcome message when not logged in", () => {
        render(
            <MemoryRouter>
                <UserProvider currentUser={null}>
                    <Home />
                </UserProvider>
            </MemoryRouter>,
        );
    
        expect(screen.getByText('DreamHost')).toBeInTheDocument();
    });
});