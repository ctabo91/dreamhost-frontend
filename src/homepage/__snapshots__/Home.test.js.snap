// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home mt-5 pt-5"
  >
    <h1
      class="Home-title display-1 text-center pt-5"
    >
      Welcome testfirst!
    </h1>
  </div>
</DocumentFragment>
`;

exports[`Home Component matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Home mt-5 pt-5"
  >
    <h1
      class="Home-title display-1 text-center pt-5"
    >
      <span>
        Welcome To
      </span>
      <p
        class="logo"
      >
        DreamHost
      </p>
    </h1>
  </div>
</DocumentFragment>
`;
